dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModelNext.java

```diff
@@ -0,0 +1,134 @@
+
+    private final Map<BigInteger, Set<LoadModel>> addressReadsMap;
+    private final Map<BigInteger, Set<StoreModel>> addressWritesMap;
+    private final Map<BigInteger, Set<MemoryEventModel>> addressAccessesMap;
+    private final Map<Thread, List<List<EventModel>>> atomicBlocksMap;
+
+    ExecutionModelNext() {
+        threadList = new ArrayList<>();
+        eventList = new ArrayList<>();
+        eventMap = new HashMap<>();
+        relationMap = new HashMap<>();
+        memoryLayoutMap= new HashMap<>();
+
+        addressReadsMap = new HashMap<>();
+        addressWritesMap = new HashMap<>();
+        addressAccessesMap = new HashMap<>();
+        atomicBlocksMap = new HashMap<>();
+    }
+
+    public void addThread(ThreadModel tModel) {
+        threadList.add(tModel);
+    }
+
+    public void addEvent(Event e, EventModel eModel) {
+        eventList.add(eModel);
+        eventMap.put(e, eModel);
+    }
+
+    public void addRelation(Relation r, RelationModel rModel) {
+        relationMap.put(r, rModel);
+    }
+
+    public void addMemoryObject(MemoryObject m, MemoryObjectModel mModel) {
+        memoryLayoutMap.put(m, mModel);
+    }
+
+    public void addAccessedAddress(BigInteger address) {
+        if (!addressAccessesMap.containsKey(address)) {
+            addressAccessesMap.put(address, new HashSet<>());
+        }
+        if (!addressReadsMap.containsKey(address)) {
+            addressReadsMap.put(address, new HashSet<>());
+        }
+        if (!addressWritesMap.containsKey(address)) {
+            addressWritesMap.put(address, new HashSet<>());
+        }
+    }
+
+    public void addAddressRead(BigInteger address, LoadModel read) {
+        addressReadsMap.get(address).add(read);
+        addressAccessesMap.get(address).add((MemoryEventModel) read);
+    }
+
+    public void addAddressWrite(BigInteger address, StoreModel write) {
+        addressWritesMap.get(address).add(write);
+        addressAccessesMap.get(address).add((MemoryEventModel) write);
+    }
+
+    public void addAtomicBlocks(Thread thread, List<List<EventModel>> atomics) {
+        atomicBlocksMap.put(thread, atomics);
+    }
+
+    public List<ThreadModel> getThreadList() {
+        return Collections.unmodifiableList(threadList);
+    }
+
+    public List<EventModel> getEventList() {
+        return Collections.unmodifiableList(eventList);
+    }
+
+    public List<EventModel> getVisibleEventList() {
+        return eventList.stream().filter(e -> e.isVisible()).toList();
+    }
+
+    public List<EventModel> getEventsByFilter(Filter filter) {
+        return eventList.stream().filter(e -> filter.apply(e.getEvent())).toList();
+    }
+
+    public EventModel getEventModelById(int id) {
+        return eventList.get(id);
+    }
+
+    public EventModel getEventModelByEvent(Event event) {
+        return eventMap.get(event);
+    }
+
+    public Set<RelationModel> getRelationModels() {
+        return new HashSet<>(relationMap.values());
+    }
```

Comment:
Normalize names (no need for "list", but use "models" consistently): getEventModels, getThreadModels, getRelationModels ...